addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/check-launch-dependency/check-launch-dependency.js

```diff
@@ -0,0 +1,433 @@
+        const resolvedInputParams = await this.resolveVarExpressions(envTypeConfig.params, resolvedVars);
+        const templateOutputs = await this.getTemplateOutputs(requestContext, envTypeId, productId);
+        const needsAlb = _.get(templateOutputs.NeedsALB, 'Value', false);
+        const maxAlbWorkspacesCount = _.get(templateOutputs.MaxCountALBDependentWorkspaces, 'Value', MAX_COUNT_ALB_DEPENDENT_WORKSPACES);
+        if (needsAlb) {
+            return await this.provisionAlb(requestContext, resolvedVars, projectId, resolvedInputParams, maxAlbWorkspacesCount);
+        }
+    }
+
+    /**
+    * Method to provision ALB. The method checks if the max count of ALB possible exists and
+    * checks if there is an ALB aready exists for the AWS account and provisions if not exists.
+    *
+    * @param requestContext
+    * @param resolvedVars
+    * @param projectId
+    * @param resolvedInputParams
+    * @param maxAlbWorkspacesCount
+    * @returns {Promise<>}
+    */
+    async provisionAlb(requestContext, resolvedVars, projectId, resolvedInputParams, maxAlbWorkspacesCount) {
+        const [ALBService] = await this.mustFindServices(['albService']);
+        const count = await ALBService.albDependentWorkspacesCount(requestContext, projectId);
+        const albExists = await ALBService.checkAlbExists(requestContext, projectId);
+        if (count >= maxAlbWorkspacesCount) {
+            throw new Error(`Error provisioning environment. Reason: Maximum workspaces using ALB has reached`);
+        }
+        if (albExists) {
+            this.print({
+                msg: `ALB Already exists for the Account. Skipping ALB Creation`,
+            });
+        } else {
+            this.print({
+                msg: `Workspace needs ALB. Provisioning an ALB.`,
+            });
+            const stackInput = await ALBService.getStackCreationInput(requestContext, resolvedVars, resolvedInputParams, projectId);
+            //Storing Dependency type so the stack completion can be handled for different dependencies
+            this.state.setKey('DEPENDENCY_TYPE', 'ALB');
+            //Create Stack
+            return await this.deployStack(requestContext, resolvedVars, stackInput);
+        }
+    }
+
+    /**
+    * Method to deploy ALB stack. The method gets cfn client and calls createStack API
+    *
+    * @param requestContext
+    * @param resolvedVars
+    * @param stackInput
+    * @returns {Promise<>}
+    */
+    async deployStack(requestContext, resolvedVars, stackInput) {
+        const cfn = await this.getCloudFormationService(requestContext, resolvedVars);
+        const response = await cfn.createStack(stackInput).promise();
+        // Update workflow state and poll for stack creation completion
+        this.state.setKey('STACK_ID', response.StackId);
+        return (
+            this.wait(5) // check every 5 seconds
+                // keep doing it for 1*1296000 seconds = 15 days
```

Comment:
Any reason we need to do it for 15 days? Does it take that long?

Reply:
We have followed the same approach as the launch-product step where the actual product is launched. Please let us know if this needs to be changed